contracts/core/Allo.sol

```diff
@@ -1,79 +1,460 @@
+    /// =========== Intializer =============
+    /// ====================================
+
+    /// @notice Initializes the contract after an upgrade
+    /// @dev During upgrade -> an higher version should be passed to reinitializer
+    /// @param _registry The address of the registry
+    /// @param _treasury The address of the treasury
+    /// @param _feePercentage The fee percentage
+    /// @param _baseFee The base fee
+    function initialize(address _registry, address payable _treasury, uint256 _feePercentage, uint256 _baseFee)
+        public
+        reinitializer(1)
+    {
+        _initializeOwner(msg.sender);
+
+        registry = Registry(_registry);
+        treasury = _treasury;
+        feePercentage = _feePercentage;
+        baseFee = _baseFee;
+
+        emit RegistryUpdated(_registry);
+        emit TreasuryUpdated(_treasury);
+        emit FeePercentageUpdated(_feePercentage);
+        emit BaseFeeUpdated(_baseFee);
+    }
+
+    /// ====================================
+    /// =========== Modifier ===============
+    /// ====================================
+
+    modifier onlyPoolOwner(uint256 _poolId) {
+        if (!hasRole(keccak256(abi.encodePacked(_poolId)), msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+        _;
+    }
+
+    /// ====================================
+    /// ==== External/Public Functions =====
+    /// ====================================
+
+    /// @notice Creates a new pool (with clone for approved strategies)
+    /// @param _identityId The identityId of the pool
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _initAllocationData The data to initialize the allocation strategy
+    /// @param _cloneAllocationStrategy Whether to clone the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _initDistributionData The data to initialize the distribution strategy
+    /// @param _cloneDistributionStrategy Whether to clone the distribution strategy
+    /// @param _token The address of the token
+    /// @param _amount The amount of the token
+    /// @param _metadata The metadata of the pool
+    /// @param _owners The owners of the pool
+    function createPoolWithClone(
+        bytes32 _identityId,
+        address _allocationStrategy,
+        bytes memory _initAllocationData,
+        bool _cloneAllocationStrategy,
+        address _distributionStrategy,
+        bytes memory _initDistributionData,
+        bool _cloneDistributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata,
+        address[] memory _owners
+    ) external payable returns (uint256 poolId) {
+        address allocationStrategy;
+        address distributionStrategy;
+
+        // DISCUSS: We either
+        // - allow cloning of every contract (use the bool)
+        // - or we force cloning of only approved contracts
```

Comment:
Is this DISCUSS still relevant or is it old? I think current implementation here is perfect, but if there's still question, let's talk about it.

Reply:
This still needs to be discussed cause the boolean arguments don't make a lot of sense at this point the way the contract is written. Looking at the code you realise, that the bool is in theory a dummy argument cause :

- if you pass bool as true but the strategy is not approved -> it reverts
- if you pass bool as true but the strategy is approved -> it clones
- if you pass bool as false but the strategy is approved -> it reverts
- if you pass bool as false but the strategy is not approved -> it doesn't clone 

This feels the  the same as 
- if contract is approved -> it clones
- if contract Is not approved -> it doesn't clone

The boolean is added just to share more clarity on reverts which I'm not sure if worth it (i could be wrong but hence the discussion )